ui/src/pages/AllStories.tsx

```diff
@@ -21,111 +85,247 @@ function createData(
+            (e) => e.ID !== action.story.ID
+          ),
+          changedVisibility: changedVisibilityContainsID
+            ? state.changedVisibility.filter((e) => e.ID !== action.story.ID)
+            : [...state.changedVisibility, action.story],
+        };
+      }
+    }
+    case "INITIALIZE_AFTER_API": {
+      return {
+        ...state,
+        visibleState: action.rows ? action.rows : [],
+        visibleTableState: action.rows ? action.rows : [],
+        tableData: action.rows ? action.rows : [],
+      };
+    }
+    case "HANDLE_CHECKED_ALL": {
+      return {
+        ...state,
+        selectedRowIds:
+          state.selectedRowIds.length === state.tableData.length
+            ? []
+            : state.tableData.map((story) => story.ID),
+      };
+    }
+    case "HANDLE_CHECKED": {
+      if (action.e.target.checked) {
+        return {
+          ...state,
+          selectedRowIds: [...state.selectedRowIds, action.story.ID],
+        };
+      } else {
+        return {
+          ...state,
+          selectedRowIds: state.selectedRowIds.filter(
+            (e) => e !== action.story.ID
+          ),
+        };
+      }
+    }
+    case "SET_ORDERING": {
+      return {
+        ...state,
+        order: action.order,
+        orderBy: action.orderBy,
+      };
+    }
+    case "SET_TABLE_DATA": {
+      return {
+        ...state,
+        tableData: action.data,
+      };
+    }
+    case "SET_TAB_VALUE": {
+      return {
+        ...state,
+        tabValue: action.newValue,
+      };
+    }
   }
+}
 
-  const [tabValue, setTabValue] = React.useState(0);
-  const [visibleState, setVisibleState] = React.useState([]);
-  const [tableData, setTableData] = useState(rows);
+export const AllStories: React.FC = () => {
+  const [state, dispatch] = useReducer(reducer, initialState);
 
-  const [selectedRowIds, setSelectedRowIds] = useState([]);
-  const [order, setOrder] = useState("desc");
-  const [orderBy, setOrderBy] = useState("id");
+  const fetchStories = (url) =>
+    fetch(url)
+      .then((res) => {
+        return res.json();
+      })
+      .then((response) =>
+        response.payload.map((story) => {
+          return createData(
+            story.ID,
+            story.title,
+            story.current_city,
+            story.year,
+            story.is_visible,
+            story.author_first_name,
+            story.author_last_name,
+            story.author_country,
+            story.image_url,
+            story.video_url,
+            story.content
+          );
+        })
+      );
 
-  const [clickedStory, setClickedStory] = useState<Story | undefined>(
+  const { data: allStories, error } = useSWR<StoryView[] | undefined>(
+    "/api/stories",
+    fetchStories
+  );
+  const [clickedStory, setClickedStory] = useState<StoryView | undefined>(
     undefined
   );
+  const classes = useStyles();
+
+  useEffect(() => {
+    dispatch({ type: "INITIALIZE_AFTER_API", rows: allStories });
```

Comment:
we should only dispatch if `allStories` is defined. Add a TODO to handle error case (if allStories stays undefined or if we get an error.